azext_iot/iothub/providers/state.py

```diff
@@ -276,22 +449,79 @@ def upload_module_identity(self, identity: dict):
 
     def delete_all_configs(self):
         configs = _iot_hub_configuration_list(target=self.target)
-        for i in tqdm(range(len(configs)), desc="Deleting configurations and edge deployments from destination hub"):
+        for i in tqdm(range(len(configs)), desc="Deleting configurations from destination hub", file=sys.stdout):
             c = configs[i]
             _iot_hub_configuration_delete(target=self.target, config_id=c["id"])
 
     def delete_all_devices(self):
         identities = _iot_device_list(target=self.target, top=-1)
-        for i in tqdm(range(len(identities)), desc="Deleting identities from destination hub"):
+        for i in tqdm(range(len(identities)), desc="Deleting devices from destination hub", file=sys.stdout):
             id = identities[i]
             _iot_device_delete(target=self.target, device_id=id["deviceId"])
 
+    def delete_all_certificates(self):
+        certificates = self.cli.invoke("iot hub certificate list --hub-name {} -g {}".format(self.hub_name, self.rg)).as_json()
+        for i in tqdm(range(len(certificates["value"])), desc="Deleting certificates from destination hub", file=sys.stdout):
+            c = certificates["value"][i]
+            self.cli.invoke("iot hub certificate delete --name {} --etag {} --hub-name {} -g {}".format(c["name"], c["etag"],
+                                                                                                        self.hub_name, self.rg))
+
+    def delete_all_endpoints(self):
+        endpoints = self.cli.invoke(f"iot hub routing-endpoint list --hub-name {self.hub_name} -g {self.rg}").as_json()
+        eventHubs = endpoints["eventHubs"]
+        serviceBusQueues = endpoints["serviceBusQueues"]
+        serviceBusTopics = endpoints["serviceBusTopics"]
+        storageContainers = endpoints["storageContainers"]
+
+        num_endpoints = len(eventHubs) + len(serviceBusQueues) + len(serviceBusTopics) + len(storageContainers)
+        pbar = tqdm(total=num_endpoints, desc="Deleting endpoints from destination hub", file=sys.stdout)
+
+        for ep in eventHubs:
+            self.cli.invoke("iot hub routing-endpoint delete --hub-name {} -g {} --endpoint-name {} --endpoint-type \
```

Comment:
just end the string and start a new one. Dont add a bunch of empty space to strings